src/counter.cpp

```diff
@@ -0,0 +1,59 @@
+#include "counter.h"
+#include <cstring>
+extern "C" {
+    #include <tree_sitter/api.h>
+}
+
+Counter::Counter(Parser* parser, Countable countable, std::string feature, std::vector<std::string>& files)
+  : parser(parser)
+  , countable(countable)
+  , feature(feature)
+  , files(files)
+{}
+
+void Counter::count_feature() {
+    for (size_t i = 0; i < files.size(); i++)
+    {
+      TSTree *tree = parser->parse_file(files[i]);
+      TSNode root_node = ts_tree_root_node(tree);
+      TSTreeCursor cursor = ts_tree_cursor_new(root_node);
+      while (true)
+      {
+          if (ts_tree_cursor_goto_first_child(&cursor) || ts_tree_cursor_goto_next_sibling(&cursor)) {
+            TSNode cur = ts_tree_cursor_current_node(&cursor);
+            const char* type = ts_node_type(cur);
+            if (!strcmp("identifier", type) && is_parent_call) {
+              is_parent_call = false;
+              uint32_t start = ts_node_start_byte(cur);
+              uint32_t end = ts_node_end_byte(cur);
```

Comment:
There's nothing wrong with using `uint32_t` here, but we can reliably expect this program to only ever be run on systems such that using `int` is fine.  It might be simpler to read if `int` is used.